src/SndFileAudioFileReader.cpp

```diff
@@ -190,3 +190,57 @@ bool SndFileAudioFileReader::run(AudioProcessor& processor)
 }
 
 //------------------------------------------------------------------------------
+
+bool SndFileAudioFileReader::configure(int channels, int samplerate, const std::string& format)
+{
+    info_.seekable = 0;
+    info_.frames = 0;
+    info_.sections = 0;
+    
+    info_.channels = channels;
+    info_.format = SF_FORMAT_RAW;
+    info_.samplerate = samplerate;
+
+    if (format == "S8") {
+        info_.format |= SF_FORMAT_PCM_S8;
+    }
+    else if (format == "U8") {
+        info_.format |= SF_FORMAT_PCM_U8;
+    }
+    else if (format == "S16LE") {
+        info_.format |= SF_FORMAT_PCM_16;
+        info_.format |= SF_ENDIAN_LITTLE;
+    }
+    else if (format == "S16BE") {
+        info_.format |= SF_FORMAT_PCM_16;
+        info_.format |= SF_ENDIAN_BIG;
+    }
+    else if (format == "S24LE") {
+        info_.format |= SF_FORMAT_PCM_24;
+        info_.format |= SF_ENDIAN_LITTLE;
+    }
+    else if (format == "S24BE") {
+        info_.format |= SF_FORMAT_PCM_24;
+        info_.format |= SF_ENDIAN_BIG;
+    }
+    else if (format == "S32LE") {
+        info_.format |= SF_FORMAT_PCM_32;
+        info_.format |= SF_ENDIAN_LITTLE;
+    }
+    else if (format == "S32BE") {
+        info_.format |= SF_FORMAT_PCM_32;
+        info_.format |= SF_ENDIAN_BIG;
+    }
+    else if (format == "FLOAT") {
```

Comment:
I think this would need to set the endianness otherwise it's determined by whatever CPU we're running on. We could have `f32be`, `f32le`, `f64be` and `f64le`?

Reply:
right! thx!